src/compiler/checker.ts

```diff
@@ -674,6 +676,12 @@ namespace ts {
             return result;
         }
 
+        function doesNameResolveToLocalSymbolForExportDefault(exports: SymbolTable, meaning: SymbolFlags, name: string): boolean {

```

Comment:
I'm thinking a simpler approach to this problem would be to have the binder just put the local name in scope as a regular symbol. Currently, for the declaration

```typescript
export default function foo() { }
```

we create a `SymbolFlags.Function` member named `default` in the `exports` symbol table, and a `SymbolFlags.ExportValue` marker member named `foo` in the `locals` symbol table. I'm thinking we could just create a local `SymbolFlags.Function` member instead, corresponding to what would happen for a normal non-exported declaration. Then we would need this awkward check.